### ISSUE TITLE
Multiline label w/ line_height < 1 renders badly

### ISSUE THREAD

**Issue by misl6** [kivy/kivy#6508]
### Versions

* Python: 3.7.1
* OS: Linux/Windows/MacOS/Android/iOS
* Kivy: master
* Kivy installation method: source code

### Description

When trying to show a multiline label w/ line_height < 1, the label renders badly (I suppose is a blending issue between the *n* layers of the `Label`).

Example screenshot (Before PR #6422 ):
![image](https://user-images.githubusercontent.com/8177736/64527571-8d930d80-d306-11e9-98b6-75b9e30b6b23.png)

Then, I submitted PR #6422, that solved the issue.

Example screenshot (After PR #6422 ):
![image](https://user-images.githubusercontent.com/8177736/64527758-0b571900-d307-11e9-9bc7-0e7d9fde02c1.png)

Unexpectedly, as @tito discovered, PR #6422 , introduced another rendering issue also on non-multiline labels (See PR #6507 ). 

### Code and Logs

Demo code for reproducing the issue:

```python
Label:
    size_hint: 1, .55
    color: 0, 0, 0, 1
    text_size: self.size
    font_size:  dp(12)
    halign: 'left'
    valign: 'top'
    text: "we need this text to wrap we need this text to wrap we need this text to wrap we need this text to wrap we need this text to wrap we need this text to wrap"
    max_lines: 3
    markup: True
    line_height: .7

```

**Comment by misl6** [comment]
As discussed on discord for future reference:

I investigated a lot about PR #6422 and #6507 and correlated issues.

Basically, `SDL_SetSurfaceBlendMode(st, SDL_BLENDMODE_BLEND)` is the right way to do it (like We're doing when there's a outline to show), but SDL leaves some dirty pixels around the glyph in certain conditions as @tito noticed.

The workaround ( as @tshirtman suggested ), could be enabling it only if the user request a line_height < 1.

**Comment by misl6** [comment]
Closed via PR #6513
